lib/src/access.ts

```diff
@@ -22,26 +32,52 @@ export type RewrapRequest = {
  * @param url Key access server rewrap endpoint
  * @param requestBody a signed request with an encrypted document key
  * @param authProvider Authorization middleware
+ * @param fulfillableObligationFQNs client-configured list of obligation value FQNs that can be fulfilled in this PEP
  * @param clientVersion
  */
 export async function fetchWrappedKey(
   url: string,
   signedRequestToken: string,
-  authProvider: AuthProvider
+  authProvider: AuthProvider,
+  fulfillableObligationFQNs: string[]
 ): Promise<RewrapResponse> {
   const platformUrl = getPlatformUrlFromKasEndpoint(url);
 
   return await tryPromisesUntilFirstSuccess(
-    () => fetchWrappedKeysRpc(platformUrl, signedRequestToken, authProvider),
+    () =>
+      fetchWrappedKeysRpc(
+        platformUrl,
+        signedRequestToken,
+        authProvider,
+        rewrapAdditionalContextHeader(fulfillableObligationFQNs)
+      ),
     () =>
       fetchWrappedKeysLegacy(
         url,
         { signedRequestToken },
-        authProvider
+        authProvider,
+        rewrapAdditionalContextHeader(fulfillableObligationFQNs)
       ) as unknown as Promise<RewrapResponse>
   );
 }
 
+/**
+ * Transform fulfillable, fully-qualified obligations into the expected KAS Rewrap 'X-Rewrap-Additional-Context' header value.
+ * @param fulfillableObligationValueFQNs
+ */
+export const rewrapAdditionalContextHeader = (
+  fulfillableObligationValueFQNs: string[]
+): string | undefined => {
+  if (!fulfillableObligationValueFQNs.length) return;
+
+  const context: RewrapAdditionalContext = {
+    obligations: {
+      fulfillableFQNs: fulfillableObligationValueFQNs.map((fqn) => fqn.toLowerCase()),
+    },
+  };
+  return base64.encode(JSON.stringify(context));
```

Comment:
`JSON.stringify` can throw.  Do we want to catch it and return undefined or re-throw a "nicer" error?